.github/workflow-pins/tests/test_check_workflow_pins.py

```
@@ -1901,6 +1901,336 @@ def test_a_flow_style_resolver_consumer_is_not_reported_dangling(self):
+
+    # ------------------------------------------------------------------
+    # Shapes the item walk must read the SAME as a line scan did (BE-8254
+    # review). Every one of these is the costly direction: an id the pre-scan
+    # misses turns a compliant workflow into a false `dangling` FAILURE, or
+    # escapes the job outright and drops a real finding.
+    # ------------------------------------------------------------------
+
+    @staticmethod
+    def _job0_step_ids(job_text):
+        lines = ("name: F\non:\n  workflow_call:\njobs:\n" + job_text).split("\n")
+        start = next(i for i, l in enumerate(lines) if l.startswith("  job0:"))
```

Ruff (0.16.2)

[error] Ambiguous variable name: `l`

(E741)